### ISSUE TITLE
[goals] Option to hide collapsed goal type display

### ISSUE THREAD

**Issue by ejgallego** [ejgallego/coq-lsp#525]
Indeed in some cases the density can become too high, requested by Anton Podkopaev (thanks!)

**Comment by bhaktishh** [comment]
Can you elaborate a bit on what this is aiming for exactly? I'm working on something that i think might be similar and planning on having it being closely integrated w coq-lsp (it's mostly in the planning stage right now but happy to chat if you're interested)

**Comment by Alizter** [comment]
Here is a screenshot demonstrating how annoying this issue is:
![image](https://github.com/ejgallego/coq-lsp/assets/8614547/b041e7ea-9425-4eaf-bae0-bad7dd40bc7d)

**Comment by ejgallego** [comment]
Yes I agree this is messy; adding an option should be super easy tho, adding a UI button to hide the goals shouldn't be too hard to.

**Comment by Alizter** [comment]
@ejgallego 

An example:
```coq
From Coq Require Import Prelude.
Inductive foo := a | b | c | d | e.

Goal forall x y z w v : foo, Type.
intros [].
- intros [].
  + intros [].
    * intros [].
      -- intros [].
      
```

Go down to the last bullet. Everything is expanded. Minimize everything. If you go up with the cursor and go down again everything is expanded again. I think the default behaviour should be minimisation.
